app/controllers/api/base_controller.rb

```diff
@@ -0,0 +1,31 @@
+class Api::BaseController < ActionController::Base
+  before_action :require_login!
+  helper_method :current_person
+
+  def require_login!
+    return true if authenticate_token
+    if token_exists
+      render json: { errors: [{ detail: 'Your key is expired' }] }, status: 401
+    else
+      render json: { errors: [{ detail: 'Access denied' }] }, status: 401
+    end
+  end
+
+  def current_person
+    @current_person ||= authenticate_token
+  end
+
+  private
+
+  def token_exists
+    authenticate_with_http_token do |token|
```

Comment:
why are you doing this `authenticate_with_http_token do |token|` 2 times?

do it one time only

in another place just check token of `current_person`